plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java

```diff
@@ -481,10 +482,24 @@ public void testRoundingOfUnspecifiedNumber()
     @Test
     public void testNumberNegativeScaleReadMapping()
     {
-        // TODO: Add similar tests for write mappings.
-        // Those tests would require the table to be created in Oracle, but values inserted
-        // by Trino, which is outside the capabilities of the current DataSetup classes.
-        SqlDataTypeTest.create()
+        addNumberNegativeScaleRoundTrips()
+                .addRoundTrip("number(37, -1)", "99999999999999999999999999999999999990", createDecimalType(38, 0), "CAST('99999999999999999999999999999999999990' AS DECIMAL(38, 0))") // max
+                .addRoundTrip("number(37, -1)", "-99999999999999999999999999999999999990", createDecimalType(38, 0), "CAST('-99999999999999999999999999999999999990' AS DECIMAL(38, 0))") // min
+                .execute(getQueryRunner(), oracleCreateAndInsert("number_negative_s"));
+    }
+
+    @Test
+    public void testNumberNegativeScaleWriteMapping()
+    {
+        addNumberNegativeScaleRoundTrips()
+                .addRoundTrip("number(37, -1)", "CAST('99999999999999999999999999999999999990' AS DECIMAL(38, 0))", createDecimalType(38, 0), "CAST('99999999999999999999999999999999999990' AS DECIMAL(38, 0))") // max
+                .addRoundTrip("number(37, -1)", "CAST('-99999999999999999999999999999999999990' AS DECIMAL(38, 0))", createDecimalType(38, 0), "CAST('-99999999999999999999999999999999999990' AS DECIMAL(38, 0))") // min
+                .execute(getQueryRunner(), oracleCreateAndTrinoInsert("number_negative_s"));
+    }
+
+    private SqlDataTypeTest addNumberNegativeScaleRoundTrips()
```

Comment:
nit: method can be `numberWithNegativeScaleTest` (see PostgreSQL's `arrayDecimalTest` etc for convention)